File: binarySearch.py

def binarySearchAppr(arr,low,high,x):
    if(low<=high):
        mid = low + (high - low)//2
        if(arr[mid]==x):
            print(f"Element is present at index {mid}")
            return
        elif(arr[mid]<x):
            return binarySearchAppr(arr,mid+1 , high, x)
        else:
            return binarySearchAppr(arr,low , mid-1, x)
            
    print("Element is not present in array")
    return 


arr = ["one", "three", "five", "seven", "nine", "eleven", "thirteen"]
x = "eleven"
arr.sort()
binarySearchAppr(arr, 0, len(arr)-1, x)

File: BinarySearch.java
//hello all
public class BinarySearch {
    public static int binarySearch(int[] arr,int k,int start,int end){

        if(start<=end){

            int mid = start+(end-start)/2;

            if(arr[mid]==k){
                return mid;
            }
            else if(arr[mid]<k){
                return binarySearch( arr, k, mid+1, end);
            }
            else{
                return binarySearch( arr, k, start, mid-1);
            }
        }
        return -1;
    }
    public static void main(String arg[]){
        //Solution ob = new Solution();
        int arr[] = {3,12,16,19,25,26,34,89,111,141};
        //Arrays.sort(arr);
        int key = 111;
        int index = BinarySearch.binarySearch(arr,key,0,arr.length-1);
        System.out.println(index);
    }
}
